File: datasets/php/HumanEval_23_strlen.php
<?php
// Return length of given string
// >>> strlen('')
// 0
// >>> strlen('abc')
// 3
function strlen($string) {
    return mb_strlen($string);
}


function candidate(...$args) {
    return strlen(...$args);
}

function test(): void {
    if (candidate('') !== 0) { throw new Exception("Test failed!"); }
    if (candidate('x') !== 1) { throw new Exception("Test failed!"); }
    if (candidate('asdasnakj') !== 9) { throw new Exception("Test failed!"); }
}

test();

File: datasets/rs-keep-code_davinci_001_temp_0.2/HumanEval_23_strlen.rs
/// Return length of given string
/// >>> strlen('')
/// 0
/// >>> strlen('abc')
/// 3
fn strlen(string: String) -> isize {
    string.len() as isize

}

fn main() {
    let candidate = strlen;
    assert_eq!(candidate(String::from("")), 0);
    assert_eq!(candidate(String::from("x")), 1);
    assert_eq!(candidate(String::from("asdasnakj")), 9);
}
